### PR TITLE
[ML] write deprecation warning when include_model_definition parameter is used

### PR DESCRIPTION
**Pull request by benwtrent**
for get trained models `include_model_definition` is now deprecated.

This commit writes a deprecation warning if that parameter is used and suggests the caller to utilize the replacement

### CODE REVIEW

**Comment by benwtrent** [comment]
run elasticsearch-ci/packaging-sample-windows

File: x-pack/plugin/src/test/resources/rest-api-spec/test/ml/inference_crud.yml [hunk 1]
```diff
@@ -871,3 +871,14 @@ setup:
   - is_false: trained_model_configs.0.estimated_heap_memory_usage
   - is_false: trained_model_configs.0.estimated_operations
   - is_false: trained_model_configs.0.license_level
+---
+"Test deprecation of include model definition param":
+  - skip:
+      features: "warnings"
+  - do:
+      warnings:
+        - "[include_model_definition] parameter is deprecated! Use parameter [include] and supply the flag 'definition' instead."
```
Comment (przemekwitek) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/ml/inference_crud.yml:880]: I'm wondering if the wording here could be more direct and say sth like: "Use [include=definition] instead".

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java [hunk 2]
```diff
@@ -62,11 +65,19 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
...
-        final GetTrainedModelsAction.Request request = restRequest.hasParam(GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION) ?
-            new GetTrainedModelsAction.Request(modelId,
+        final GetTrainedModelsAction.Request request;
+        if (restRequest.hasParam(GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION)) {
+            deprecationLogger.deprecate(
+                GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION,
+                "[{}] parameter is deprecated! Use parameter [{}] and supply the flag 'definition' instead.",
+                GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION,
+                GetTrainedModelsAction.Request.INCLUDE.getPreferredName());
+            request = new GetTrainedModelsAction.Request(modelId,
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java:75]: Should this constructor now go away?
```
        public Request(String id, boolean includeModelDefinition, List<String> tags) {
```

Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java:75]: No, it's a convenience method for when the user calls with a deprecated flag. Consequently, itself is also deprecated.

File: x-pack/plugin/src/test/resources/rest-api-spec/test/ml/inference_crud.yml (hunk 1 above)
Comment (benwtrent) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/ml/inference_crud.yml:880]: I can update the message, for sure.

**Review by przemekwitek — APPROVED** [review]
LGTM

Changes in the message do not require further review from my side.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java [hunk 3]
```diff
@@ -62,11 +65,19 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
                 restRequest.paramAsStringArray(
                     GetTrainedModelsAction.Request.INCLUDE.getPreferredName(),
                     Strings.EMPTY_ARRAY)));
-        final GetTrainedModelsAction.Request request = restRequest.hasParam(GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION) ?
-            new GetTrainedModelsAction.Request(modelId,
+        final GetTrainedModelsAction.Request request;
+        if (restRequest.hasParam(GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION)) {
+            deprecationLogger.deprecate(
+                GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION,
+                "[{}] parameter is deprecated! Use parameter [{}] and supply the flag 'definition' instead.",
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java:72]:
```suggestion
                "[{}] parameter is deprecated! Use [include=definition] instead.",
```

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java [hunk 4]
```diff
@@ -62,11 +65,19 @@ protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient
...
+                GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION,
+                GetTrainedModelsAction.Request.INCLUDE.getPreferredName());
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/inference/RestGetTrainedModelsAction.java:74]:
```suggestion
                "[{}] parameter is deprecated! Use [include=definition] instead.",
                GetTrainedModelsAction.Request.INCLUDE_MODEL_DEFINITION);
```

File: x-pack/plugin/src/test/resources/rest-api-spec/test/ml/inference_crud.yml (hunk 1 above)
Comment (benwtrent) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/ml/inference_crud.yml:880]:
```suggestion
        - "[include_model_definition] parameter is deprecated! Use [include=definition] instead."
```

**Pull request merged**